pkg/controller/util.go

```diff
@@ -959,3 +1022,54 @@ func getCertConfigMap(client kubernetes.Interface, pvc *v1.PersistentVolumeClaim
 
 	return value, nil
 }
+
+func getURLFromIngress(ing *extensionsv1beta1.Ingress, uploadProxyServiceName string) string {
+	if ing.Spec.Backend != nil {
+		if ing.Spec.Backend.ServiceName != uploadProxyServiceName {
+			return ""
+		}
+		return ing.Spec.Rules[0].Host
+	}
+	for _, rule := range ing.Spec.Rules {
+		for _, path := range rule.HTTP.Paths {
+			if path.Backend.ServiceName == uploadProxyServiceName {
+				if rule.Host != "" {
+					return rule.Host
+				}
+			}
+		}
+	}
+	return ""
+
+}
+
+func getURLFromRoute(route *routev1.Route, uploadProxyServiceName string) string {
+	if route.Spec.To.Name == uploadProxyServiceName {
+		if route.Spec.Host != "" {
+			return route.Spec.Host
```

Comment:
I wonder about getting this value from the `spec` of the Route.  I would think that something from `status` would be more appropriate.  Like `status.ingress[].host`?

Reply:
done